Code/Framework/AzFramework/AzFramework/Visibility/OcclusionBus.h

```diff
@@ -39,22 +40,49 @@ namespace AzFramework
+            const AZ::Name& viewName, const AZ::Vector3& cameraWorldPos, const AZ::Matrix4x4& cameraWorldToClip) = 0;
 
         //! Returns true if the entity is not occluded in the occlusion view. Otherwise, returns false.
-        virtual bool IsEntityVisibleInOcclusionView(const AZ::Name& name, const AZ::EntityId& entityId) const = 0;
+        virtual bool GetOcclusionViewEntityVisibility(const AZ::Name& viewName, const AZ::EntityId& entityId) const = 0;
 
         //! Returns true if the AABB is not occluded in the occlusion view. Otherwise, returns false.
-        virtual bool IsAabbVisibleInOcclusionView(const AZ::Name& name, const AZ::Aabb& bounds) const = 0;
+        virtual bool GetOcclusionViewAabbVisibility(const AZ::Name& viewName, const AZ::Aabb& bounds) const = 0;
+
+        //! @brief Test if target shapes are visible from the perspective of a source shape, within a volume extruded between them.
+        //! @param sourceAabb Bounding volume for the source shape defining the frame of reference for visibility tests.
+        //! @param targetAabbs Vector of bounding volumes for target shapes that will be tested for visibility.
+        //! @return A vector of flags containing the visibility state for corresponding target shapes. The size of the returned container
+        //! should match the size of the input target container unless an ever occurred.
+        virtual AZStd::vector<bool> GetOcclusionViewAabbToAabbVisibility(
+            const AZ::Name& viewName, const AZ::Aabb& sourceAabb, const AZStd::vector<AZ::Aabb>& targetAabbs) const = 0;
+
+        //! @brief Test if target shapes are visible from the perspective of a source shape, within a volume extruded between them.
+        //! @param sourceSphere Bounding volume for the source shape defining the frame of reference for visibility tests.
+        //! @param targetSpheres Vector of bounding volumes for target shapes that will be tested for visibility.
+        //! @return A vector of flags containing the visibility state for corresponding target shapes. The size of the returned container
+        //! should match the size of the input target container unless an ever occurred.
+        virtual AZStd::vector<bool> GetOcclusionViewSphereToSphereVisibility(
+            const AZ::Name& viewName, const AZ::Sphere& sourceSphere, const AZStd::vector<AZ::Sphere>& targetSpheres) const = 0;
+
+        //! @brief Test if target entities are visible from the perspective of a source entity, within a volume extruded between them.
+        //! @param sourceEntityId Id for the source entity defining the frame of reference for visibility tests.
+        //! @param targetEntityIds Vector of Ids for target entities that will be tested for visibility.
+        //! @return A vector of flags containing the visibility state for corresponding target entities. The size of the returned container
+        //! should match the size of the input target container unless an ever occurred.
```

Comment:
```suggestion
        //! should match the size of the input target container unless an error occurred, in which case an empty vector is returned. 
```
Or we use AZ::Outcome for the result type